### PR TITLE
Moved core & ui to subfolders

### PR DESCRIPTION
**Pull request by ianmcburnie**
Moved core & ui to subfolders.

Various changes to package scripts were also required.

Relative paths were updated in docs files.

EDIT: Also incudes fix for #163. I made a boo boo and this branch started with the changes in this branch for makeup-tabs: https://github.com/makeup/makeup-js/commit/348b366b82701606be2152d65b2abdf7b7df54ae#diff-b04b38d4e36f7a7171aeb211bf933eaf36d41d9866ebbc3639f673f84dc350ae For some reason I thought I'd pushed those changes to master (I even closed the issue).

### CODE REVIEW

**Comment by patricknelson** [comment]
Just gonna note that work on this PR also relates to https://github.com/makeup/makeup-js/issues/153#issuecomment-2011171334. 🙂 

Overall this looks great since a lot of little bits and bobs were cleaned up. This also offered the opportunity to spot a few minor inconsistencies which I flagged in the review (mostly just missing links). 

p.s. After the tedium of double checking which packages are properly documented in their corresponding folders, it may make sense to also script that as well. Maybe all that's needed is just a simple report of what's in sync or not, which could probably be done with a quick bash script. Basically what I did was:

* Highlight differences between doc folders and package folders (e.g. package exists but doc doesn't, or doc exists but package doesn't)
* Determine if a doc folder has an `index.html` file
* Determine if the doc index properly links all doc folders below it (could itself move to being auto-generated by just iterating over subdirs, but for just kept manual...)

I found I could relatively easily script those first two things. See `./package-report.sh` in this commit on my work (since I still don't have push access): https://github.com/makeup/makeup-js/commit/93dc7ae9867c103391ab6610a36d0754073b77ad. It's in the root right now but that could be changed easily (would just need to adjust how `$BASE_DIR` is determined).

**Comment by patricknelson** [comment]
Found another issue: All of the `icons.svg` in the docs are 404'ing right now. Just need to update the `src` to `../../icons.svg`.

File: docs/ui/index.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <li><a href="makeup-fullscreen-dialog/index.html">makeup-fullscreen-dialog</a></li>
+        <li><a href="makeup-lightbox-dialog/index.html">makeup-lightbox-dialog</a></li>
+        <li><a href="makeup-listbox/index.html">makeup-listbox</a></li>
+        <li><a href="makeup-listbox-button/index.html">makeup-listbox-button</a></li>
+        <li><a href="makeup-menu/index.html">makeup-menu</a></li>
+        <li><a href="makeup-menu-button/index.html">makeup-menu-button</a></li>
+        <li><a href="makeup-panel-dialog/index.html">makeup-panel-dialog</a></li>
+        <li><a href="makeup-snackbar-dialog/index.html">makeup-snackbar-dialog</a></li>
+        <li><a href="makeup-switch/index.html">makeup-switch</a></li>
+        <li><a href="makeup-tabs/index.html">makeup-tabs</a></li>
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply docs/ui/index.html:37]: Oops. I have all the makeup-tabs work in a git stash on local. I could have sworn I'd pushed it!! Argh. I'll resolve it.

File: docs/core/index.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      <p>The following modules assist with common accessibility logic (e.g. maintaining a roving tabindex).</p>
+      <ul>
+        <li><a href="makeup-active-descendant/index.html">makeup-active-descendant</a></li>
+        <li><a href="makeup-exit-emitter/index.html">makeup-exit-emitter</a></li>
+        <li><a href="makeup-expander/index.html">makeup-expander</a></li>
+        <li><a href="makeup-floating-label/index.html">makeup-floating-label</a></li>
+        <li><a href="makeup-focusables/index.html">makeup-focusables</a></li>
+        <li><a href="makeup-key-emitter/index.html">makeup-key-emitter</a></li>
+        <li><a href="makeup-keyboard-trap/index.html">makeup-keyboard-trap</a></li>
+        <li><a href="makeup-modal/index.html">makeup-modal</a></li>
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply docs/core/index.html:26]: Actually, modal is just a wrapper around two other core modules, so it should be in core.

File: docs/core/index.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <body>
+    <main role="main">
+      <h1>Core Modules</h1>
+
+      <p>The following modules assist with common accessibility logic (e.g. maintaining a roving tabindex).</p>
+      <ul>
+        <li><a href="makeup-active-descendant/index.html">makeup-active-descendant</a></li>
+        <li><a href="makeup-exit-emitter/index.html">makeup-exit-emitter</a></li>
+        <li><a href="makeup-expander/index.html">makeup-expander</a></li>
+        <li><a href="makeup-floating-label/index.html">makeup-floating-label</a></li>
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply docs/core/index.html]: Fixed.

File: docs/ui/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (patricknelson) [reply docs/ui/index.html:39]: LGTM

**Pull request merged**